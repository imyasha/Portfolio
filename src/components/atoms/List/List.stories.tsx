import { List } from './List';

export default {
	title: 'atoms/List',
	component: List,
};

export const ListStory = ({ children }: any) => <List>{children}</List>;

ListStory.storyName = 'List';
ListStory.args = {
	children: <li>List element</li>,
};
