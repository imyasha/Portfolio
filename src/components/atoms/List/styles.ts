import styled from 'styled-components';

const StyledList = styled.ul`
	margin: 0;
	list-style: square;
	padding-left: 1rem;

	li {
		margin-bottom: 0.5rem;
	}
`;

export { StyledList };
