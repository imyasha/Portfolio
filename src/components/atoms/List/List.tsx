import { FC } from 'react';
import { StyledList } from './styles';

const List: FC = ({ children }) => {
	return <StyledList>{children}</StyledList>;
};

export { List };
