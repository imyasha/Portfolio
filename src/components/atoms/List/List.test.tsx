import { List } from '../List';
import { render } from '@testing-library/react';

describe('List', () => {
	it('renders correctly', () => {
		const { container } = render(
			<List>
				<li>List element</li>
			</List>
		);
		expect(container).toMatchSnapshot();
	});
});
