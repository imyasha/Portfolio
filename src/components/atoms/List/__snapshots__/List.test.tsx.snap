// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <ul
    class="styles__StyledList-sze0tr-0 fqIYCz"
  >
    <li>
      List element
    </li>
  </ul>
</div>
`;
